index.js

```diff
@@ -9,17 +9,14 @@ const create = prod => (start, sink) => {
   let got2;
   let unsub;
   sink(0, t => {
-    got2 = got2 || t === 2;
+    end = end || t === 2;
```

Comment:
assign this function to a const ask = t => {...} and pass it to sink(0, ask). Then inside the producer, after sink(t, d); just call ask(t);

Reply:
done, although I've named it `maybeDispose` (not the best name, but I dont feel `ask` is appropriate here - its used often in sinks when they want to "ask"/pull for data which is not the case here), any suggestions on how to rename this?